manifests/params.pp

```diff
@@ -4,30 +4,30 @@
   $ssl_dir = '/etc/puppetlabs/puppet/ssl'
 
   # Set CA file
-  if defined('$::rh_certificate_repo_ca_file') {
-    $rh_certificate_repo_ca_file = $::rh_certificate_repo_ca_file
+  if defined('$facts[rh_certificate_repo_ca_file]') {
+    $rh_certificate_repo_ca_file = $facts['rh_certificate_repo_ca_file']
   }
   else {
     $rh_certificate_repo_ca_file = undef
   }
 
   # Set host cert
-  if defined('$::rh_certificate_consumer_host_cert') {
-    $rh_certificate_consumer_host_cert = $::rh_certificate_consumer_host_cert
+  if defined('$facts[rh_certificate_consumer_host_cert]') {
+    $rh_certificate_consumer_host_cert = $facts['rh_certificate_consumer_host_cert']
   }
   else {
     $rh_certificate_consumer_host_cert = undef
   }
 
```

Comment:
Technically, this can be folded down to

```ruby
$rh_certificate_consumer_host_cert = $facts['rh_certificate_consumer_host_cert']
```

as it will become `undef` automatically if the fact is not defined, right?

Reply:
At some point there was a movement to make hashes work more like Python where missing keys were a problem (I think `strict_variables` still works that way), but I think that died out and today you're right that you can do that. Otherwise you also have `fact()` in stdlib which for sure accepts it. It also accepts dot-notation so you can use `fact('some.nested.fact')` and it returns undef. Then again, you can also use `$facts.dig('some', 'nested', 'fact')`.